OpenRA.Mods.Common/Widgets/TooltipContainerWidget.cs

```diff
@@ -25,23 +25,46 @@ public class TooltipContainerWidget : Widget
 
 		public Action BeforeRender = Nothing;
 		public int TooltipDelayMilliseconds = 200;
+		readonly Widget container;
 		Widget tooltip;
 		int nextToken = 1;
 		int currentToken;
+		string id;
+		WidgetArgs widgetArgs;
 
 		public TooltipContainerWidget()
 		{
+			container = new ContainerWidget();
+			AddChild(container);
+
 			graphicSettings = Game.Settings.Graphics;
-			IsVisible = () => Game.RunTime > Viewport.LastMoveRunTime + TooltipDelayMilliseconds;
+			IsVisible = () =>
+			{
+				// PERF: Only load widget once visible.
+				var visible = Game.RunTime > Viewport.LastMoveRunTime + TooltipDelayMilliseconds;
+				if (visible)
+					LoadWidget();
+
+				return visible;
+			};
+		}
+
+		void LoadWidget()
+		{
+			if (id == null || tooltip != null)
+				return;
+
+			tooltip = Ui.LoadWidget(id, container, new WidgetArgs(widgetArgs) { { "tooltipContainer", this } });
+			widgetArgs = null;
```

Comment:
What is the reason for `null`ing this out?

Reply:
Only to relase the reference to it. Perhaps not needed.

Reply:
I don't think this is necessary then.

Reply:
Removed.